build/package/scripts/cache-build.sh

```diff
@@ -1,90 +1,148 @@
-#!/bin/bash
+#!/usr/bin/env bash  
+# avoids ancient bash on macos
 set -eu
 # Copy build to cache
 
-
 # the copy commands are based on GNU cp tools
 # On a mac `brew install coreutils` gives `g` prefixed cmd line tools such as gcp
 # to use these define env variable GNU_CP=gcp before invoking this script.
 CP="${GNU_CP:-cp}"
 LS="${GNU_LS:-ls}"
 
-OUTPUT_DIR="docker"
-WORKING_DIR="."
-CACHE_BUILD_KEY=
-CACHE_LOCATION_USED_PATH=
-DEBUG="${DEBUG:-false}"
+join() {
+  local IFS="$1"
+  shift
+  echo "$*"
+}
+
+splitAtColon() { 
+  # colon is 
+  echo "$1" | tr ":" "\n"
+}
+# https://stackoverflow.com/a/918931
+
+outputs_str=""
+extra_inputs_str=""
+declare -a outputs
+outputs=()
+declare -a inputs
+inputs=()
+working_dir=.
+cache_build_key=
+cache_location_used_path=
+debug="${DEBUG:-false}"
+dry_run=false
+
 
 while [[ "$#" -gt 0 ]]; do
     case $1 in
 
-    --working-dir) WORKING_DIR="$2"; shift;;
-    --working-dir=*) WORKING_DIR="${1#*=}";;
+    --working-dir) working_dir="$2"; shift;;
+    --working-dir=*) working_dir="${1#*=}";;
+
+    --cached-outputs) outputs_str="$2"; shift;;
+    --cached-outputs=*) outputs_str="${1#*=}";;
+
+    --cache-extra-inputs) extra_inputs_str="$2"; shift;;
```

Comment:
I find the name a bit confusing. Maybe something like `--cache-dependencies`? Though I am not convinced this is really better. Or maybe just changing the order to `--extra-cache-inputs` would help? If you do not like any of these suggestions I'm fine with what you propose.

Reply:
As mentioned in the other comment I called now the combination of what previously would have been `working-dir` and then the `cache-extra-inputs` as `cache-sources`. Let me know if you find this confusing or have a worthwhile alternative proposal.